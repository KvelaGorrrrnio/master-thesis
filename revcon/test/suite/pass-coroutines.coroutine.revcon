@coroutine

coroutine main(int a) {
  spawn w1 worker
  spawn h1 handler
  resume h1 (w1,a)
  despawn h1 handler
  despawn w1 worker unique
}

coroutine worker(int x) {
  x += 100
  yield unique
}

coroutine handler(coroutine(int) c, int x) {
  resume c (x)
}
