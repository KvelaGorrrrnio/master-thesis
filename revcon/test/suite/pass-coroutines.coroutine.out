test/suite/pass-coroutines.coroutine.revcon:
[store]
  a: 100
  main: Ã˜

[stack]
[spawn main main begin () | resume main (a) | despawn main main end ()] <|> []

